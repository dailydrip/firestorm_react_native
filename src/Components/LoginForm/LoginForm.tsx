import React, { Fragment } from 'react'
import { View, Image } from 'react-native'
import styles from './LoginFormStyle'
import { FormLabel, FormInput, FormValidationMessage, Button, Text } from 'react-native-elements'
import Colors from '../../Config/Colors'

type Props = {
  email: string
  password: string
  setEmail: (email: String) => void
  setPassword: (password: String) => void
  authenticate: () => void
}

export default class LoginForm extends React.PureComponent<Props> {
  render () {
    const { email, password, setEmail, setPassword, authenticate } = this.props
    return (
      <Fragment>
        <View style={{ height: '40%', padding: 50 }}>
          <Image
            source={{ uri: 'https://raw.githubusercontent.com/dailydrip/firestorm/master/assets/static/images/firestorm-logo.png' }}
            resizeMode='contain'
            style={{ height: '100%', width: '100%' }} />
        </View>

        <FormLabel>E-mail</FormLabel>
        <FormInput autoCapitalize='none' value={email} onChangeText={email => setEmail(email)} />

        <FormLabel>Password</FormLabel>
        <FormInput value={password} secureTextEntry onChangeText={password => setPassword(password)} />

        <Button backgroundColor={Colors.action} onPress={() => authenticate()} large style={{ marginTop: 30 }} title='Login' />
      </Fragment>
    )
  }
}
