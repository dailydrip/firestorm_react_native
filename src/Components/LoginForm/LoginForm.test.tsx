import React from 'react'
import LoginForm from './LoginForm'
import renderer from 'react-test-renderer'

describe('Login Form Component test', () => {
  it('should render', () => {
    const tree = renderer
      .create(<LoginForm authenticate={() => null} email='' password='' setEmail={() => null} setPassword={() => null} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Test', () => {
  it('works', () => {
    expect(true).toBe(true)
  })
})
