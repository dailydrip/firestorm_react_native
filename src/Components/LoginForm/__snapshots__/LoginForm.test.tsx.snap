// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component test should render 1`] = `
Array [
  <View
    style={
      Object {
        "height": "40%",
        "padding": 50,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://raw.githubusercontent.com/dailydrip/firestorm/master/assets/static/images/firestorm-logo.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      E-mail
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>,
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Password
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "color": "#86939e",
            "minHeight": 36,
            "width": 750,
          },
          Object {
            "fontSize": 17.5,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginTop": 30,
        }
      }
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#FAAF3F",
            },
            undefined,
            false,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>,
]
`;
